### ISSUE TITLE
Error: failed to load configuration: failed to create store: unable to load on store creation: invalid config: Config.IsValid: model.config.is_valid.site_url.app_error

### ISSUE THREAD

**Issue by cristiansanduchub** [mattermost/docker#41]
When trying to start the container with or without NGINX, I get this error:
Error: failed to load configuration: failed to create store: unable to load on store creation: invalid config: Config.IsValid: model.config.is_valid.site_url.app_error

**Comment by rubiusus** [comment]
> I had the same issues and the fault are in some variables in .env file. The docker-compose can't resolve the variables that are used in other variables in the .env file:
> 
> For example, in the .env file:
> 
> ```
> DOMAIN=my-domain.com
> MM_SERVICESETTINGS_SITEURL=https://${DOMAIN}  # <-- can't resolve DOMAIN
> 
> # The same error here, can't resolve POSTGRES_USER, POSTGRES_PASSWORD and POSTGRES_DB:
> MM_SQLSETTINGS_DATASOURCE=postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@postgres:5432/${POSTGRES_DB}?sslmode=disable&connect_timeout=10
> ```
> 
> You can add these lines in the docker-compose.yml directly:
> 
> ```dockerfile
> environment:
>   - MM_SQLSETTINGS_DATASOURCE=postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@postgres:5432/${POSTGRES_DB}?sslmode=disable&connect_timeout=10
>   - MM_SERVICESETTINGS_SITEURL=https://${DOMAIN}
> ```

I have the same issue but I don't understand if I am trying to fix it correctly. I have added those lines in docker-compose.yml directly but with no success. Maybe I have to check something else?

**Comment by ljgago** [comment]
The issue is that the docker-compose.yml can't resolve nested environment variables, e.g.:

```
# .env file:
DOMAIN=my-domain.com
MM_SERVICESETTINGS_SITEURL=https://${DOMAIN}

# docker-compose-yml
environment:
  - MM_SERVICESETTINGS_SITEURL
```

`MM_SERVICESETTINGS_SITEURL` depend of `DOMAIN` and docker-compose can't resolve `DOMAIN`, and the result is `https://${DOMAIN}` all as string.

You need to write environment variables in the .env without depend on others.

If I write this in the docker-compose.yml, I depend on a single variable:

```
# docker-compose.yml
environment:
  - MM_SERVICESETTINGS_SITEURL=https://${DOMAIN}
```

**Comment by rubiusus** [comment]
Thank you @ljgago for you further explanation. Now I've got it and I could fix the docker-compose.yml file.
Mattermost container works now. :)
I have still a problem with the postgres container ( error "mattermost db does not exist") I'll open a new thread for that.

Thank you

**Comment by mrckndt** [comment]
> The issue is that the docker-compose.yml can't resolve nested environment variables

It can. Since version 1.26 (or later, i forgot the precise version) `docker-compose` can interpolate nested variables. Because of that a newer `docker-compose` is a prequisite https://github.com/mattermost/docker#prequisites.

Tl;dr: use a newer `docker-compose` version.

**Comment by advra** [comment]
Hello I faced the same issue. If you cannot update `docker-compose` then in the mean time you can edit the - `MM_SERVICESETTINGS_SITEURL=https://${DOMAIN}` without variable interpretation and hard code the domain with `MM_SERVICESETTINGS_SITEURL=https://mm.example.com` instead.
